src/Microsoft.AspNetCore.Mvc.TagHelpers/Internal/CryptographyAlgorithms.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Security.Cryptography;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers.Internal
+{
+    public static class CryptographyAlgorithms
+    {
+        public static SHA256 CreateSHA256()
+        {
+            SHA256 sha256;
+
+#if NETSTANDARD1_6
+            sha256 = SHA256.Create();
+#else
+            try
+            {
+                sha256 = SHA256.Create();
+            }
+            catch (System.Reflection.TargetInvocationException)
```

Comment:
This appears to be the documented exception type that is thrown: https://msdn.microsoft.com/en-us/library/z08hz7ad(v=vs.110).aspx

But that's pretty ugly code... not sure if there's anything better we can do here...